### DISCUSSION TITLE
Raw string option but still allow for pattern to be included

### DISCUSSION THREAD

**Discussion by asmodeus812** [BurntSushi/ripgrep#2613] [Q&A] [answered]
I would like to ask if there is an option similar to -F raw string option, but still be able to add escaped pattern to the string, the reason being is that more often than not we want to search by raw string directly, and sometimes a pattern or atom is required.  I was aiming for a behavior somewhat similar to the default way searching is done in vim with / or ? Where if you do not add \v or \m the string is taken literally, but you could still add escape characters such as backslash to specify regexp atoms - something like that - `rawstring.that.contains()special-characters\.\*`

**Comment by BurntSushi** [comment, marked as answer]
I think this is basically what BREs. And yes, vim supports a flavor of this. I am personally not a huge fan because I think the multiple flavors ends up making things pretty confusing.

**Comment by asmodeus812** [comment]
@BurntSushi on my side i am not really interested in at all supporting the magic flags such as \v and \m, but simply extending the -F option, or have another option to combine with -F which will allow us to escape atoms in the string, it's the reverse of the default behavior, without any flags or args, i guess, where everything will be taken literally as text, unless escaped.

**Reply by BurntSushi** [reply]
Yes, I understand. I personally just don't want to introduce another flavor of syntax.
